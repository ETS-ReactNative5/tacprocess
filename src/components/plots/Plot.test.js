import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Plot from "./Plot";

const caption ="The Plot Caption";

// Checking if the Plot Component renders correctly for different inputs
describe("Plot Component", () => {

  //Enzyme method testing if it renders correctly with empty value of the target
  it("Render the Plot Component having unpopulated props with no errors", () => {
    const rendered = mount(<Plot caption={""} />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });

  it("Render the Plot Component having populated props with no errors", () => {
    const rendered = mount(<Plot caption={caption} />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });

});
