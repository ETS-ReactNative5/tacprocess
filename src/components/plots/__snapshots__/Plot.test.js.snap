// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plot Component Render the Plot Component having populated props with no errors 1`] = `
<Plot
  caption="The Plot Caption"
>
  <div
    className="stat-item"

  >
    <div>
      <PlotCaption
        caption="The Plot Caption"
      >
        <div
          className="plot-caption"
          dangerouslySetInnerHTML={
            Object {
              "__html": "The Plot Caption",
            }
          }
        />
      </PlotCaption>
    </div>
  </div>
</Plot>
`;

exports[`Plot Component Render the Plot Component having unpopulated props with no errors 1`] = `
<Plot
  caption=""
>
  <div
    className="stat-item"

  >
    <div />
  </div>
</Plot>
`;
